tests/stats_test.cpp

```diff
@@ -0,0 +1,415 @@
+
+static std::shared_ptr<nodes_data> global_data;
+
+static void destroy_global_data()
+{
+	global_data.reset();
+}
+
+class test_event : public ioremap::cache::treap_node_t<test_event> {
+public:
+	test_event() = default;
+	test_event(const std::string& id, uint64_t size, double frequency, time_t last_access)
+	: m_id(id), m_size(size), m_frequency(frequency), m_last_access(last_access)
+	{}
+
+	uint64_t get_weight() const { return m_size; }
+	void set_weight(uint64_t weight) { m_size = weight; }
+
+	double get_frequency() const { return m_frequency; }
+	void set_frequency(double freq) { m_frequency = freq; }
+
+	time_t get_time() const {return m_last_access; }
+	void set_time(time_t time) { m_last_access = time; }
+
+	inline static bool key_compare_event(const test_event &lhs, const test_event &rhs) {
+		return key_compare(lhs.get_key(), rhs.get_key()) < 0;
+	}
+
+	inline static bool weight_compare(const test_event &lhs, const test_event &rhs) {
+		return lhs.get_weight() < rhs.get_weight();
+	}
+
+	// treap_node_t
+	typedef ioremap::cache::treap_node_traits<test_event>::key_type key_type;
+	typedef ioremap::cache::treap_node_traits<test_event>::priority_type priority_type;
+
+	key_type get_key() const { return &m_id; }
+	priority_type get_priority() const { return m_last_access; }
+
+	inline static int key_compare(const key_type &lhs, const key_type &rhs) {
+		return lhs->compare(*rhs);
+	}
+
+	inline static int priority_compare(const priority_type &lhs, const priority_type &rhs) {
+		if (lhs < rhs) {
+			return 1;
+		}
+
+		if (lhs > rhs) {
+			return -1;
+		}
+
+		return 0;
+	}
+
+private:
+	std::string		m_id;
+	uint64_t		m_size;
+	double			m_frequency;
+	time_t			m_last_access;
+};
+
+#define TOP_LENGTH 50
+#define EVENTS_LIMIT 1000
+#define EVENTS_SIZE (static_cast<int64_t>(EVENTS_LIMIT * sizeof(test_event)))
+#define PERIOD_IN_SECONDS 300
+
+static void configure_nodes(const std::string &path)
+{
+	config_data top_params = config_data()
+		("top_length", TOP_LENGTH)
+		("events_size", EVENTS_SIZE)
+		("period_in_seconds", PERIOD_IN_SECONDS);
+
+	start_nodes_config start_config(results_reporter::get_stream(), std::vector<server_config>({
+		server_config::default_value().apply_options(config_data()
+			("group", 5)
+			("monitor_top", top_params)
+		)
+	}), path);
+
+	global_data = start_nodes(start_config);
+}
+
+/*
+ * Top statistics handler must exist, if "top" section in "monitor" section in config
+ * exists. Node's configuration happened in configure_node() function above.
+ */
+static void test_top_statistics_existence()
+{
+	dnet_node *node = global_data->nodes[0].get_native();
+	auto monitor = ioremap::monitor::get_monitor(node);
+	BOOST_CHECK(monitor != nullptr);
+
+	auto top_stats = monitor->get_top_stats();
+	BOOST_CHECK(top_stats != nullptr);
+}
+
+/****************
+ Test event_stats
+ ****************/
+typedef ioremap::monitor::event_stats<test_event> stats_t;
+
+static void test_event_stats_boundary_conditions()
```

Comment:
Please describe these tests with more details, similar to what I have asked for python test